### PR TITLE
Reviews

### PR DESCRIPTION
**Pull request by Christiandecolombia**
Created functionality for review
controller, repo, service, model. 
the front-end will be completed in the next story.

### CODE REVIEW

File: src/main/frontend/components/ReviewTile.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  const{user, review, rating} = props.review;
+
+  return(
+    <div>
+      <p>{user} {rating}</p>
+      <p>{review}</p>
+    </div>
+  )
+}
+
```
Comment (safang99) [inline src/main/frontend/components/ReviewTile.js:15]: Looks like this needs a component which is using the `ReviewTile`

File: src/main/java/com/launchacademy/reviews/controllers/ApiV1ReviewsController.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    Map<String, Review> persistedReview = new HashMap<>();
+    review.setUserName(newReview.get("name"));
+    review.setRating(Integer.valueOf(newReview.get("rating")));
+    review.setReview(newReview.get("review"));
+    review.setLongestRelationship(Integer.valueOf(newReview.get("longestRelationship")));
+    reviewService.save(review);
+    persistedReview.put("review", review);
+
+    return persistedReview;
+  }
```
Comment (safang99) [inline src/main/java/com/launchacademy/reviews/controllers/ApiV1ReviewsController.java]: I see the `POST` endpoint has been added here.  Is the goal to add a form on the front end?

File: src/main/java/com/launchacademy/reviews/models/Review.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  @Column(name = "review")
+  private String review;
+
+  @Column(name="logest_relationship")
+  private Integer longestRelationship;
+
+//  @ManyToOne
+//  @JsonIgnoreProperties("reviews")
+//  @JoinColumn(name = "site_id")
+//  private Site site;
```
Comment (safang99) [inline src/main/java/com/launchacademy/reviews/models/Review.java]: It looks like we will want to bring this back in if there is a one to many relationship to establish between `sites`

File: src/main/java/com/launchacademy/reviews/seeders/ReviewSeeder.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+  public ReviewSeeder(ReviewService reviewService, SiteService siteService) {
+    this.reviewService = reviewService;
+    this.siteService = siteService;
+  }
+
+  public void seed(){
+    if(reviewService.findAll().isEmpty()){
+      Site site = siteService.findByName("").get(0);
+    }
```
Comment (safang99) [inline src/main/java/com/launchacademy/reviews/seeders/ReviewSeeder.java:262]: After finding a site we can add some reviews to it

**Review by safang99 — CHANGES_REQUESTED** [review]
I know you mentioned that this is just the backend and you are building out the frontend in the next PR, but I would like to see the complete feature in one PR unless there is a reason you are trying to get this merged now.  I am a little concerned of the scope of this PR due to the pieces I am seeing so let's sync on the story you are working on

**Comment by Christiandecolombia** [comment]
Finished Review feature.
Reviews are seeded and displaying on the site show page.

File: src/main/frontend/components/NewReviewForm.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+              type="number"
+              value={formPayload.longestRelationship}
+              onChange={handleInputChange}
+          />
+        </div>
+        <input className="button" type="submit" value="Submit" />
+      </form>
+  )
+}
+export default NewReviewForm
```
Comment (safang99) [inline src/main/frontend/components/NewReviewForm.js]: Indentation looks to be at a size of 4, but we will want to use 2 to be consistent with the rest of the code.

Also for readability we will want to use an empty line to separate the end of one function from the start of the next.  For example 29&30, 36&37, etc.

File: src/main/frontend/components/SitesIndex.js [hunk 6]
```diff
@@ -8,6 +8,7 @@ const SitesIndex = props => {
   const fetchSites = async () => {
     try {
       const response = await fetch("api/v1/sites")
+      console.log("inside index fetch")
```
Comment (safang99) [inline src/main/frontend/components/SitesIndex.js]: We can remove this log statement

File: src/main/frontend/components/SitesIndex.js [hunk 7]
```diff
@@ -34,6 +35,7 @@ const SitesIndex = props => {
 
   return (
     <div>
+      <p>this is site index</p>
```
Comment (safang99) [inline src/main/frontend/components/SitesIndex.js]: I might go with something different for the title of this page.  Index/Show/New may mean alot to us, but it won't mean much to the users visiting your website.

File: src/main/java/com/launchacademy/reviews/controllers/ApiV1ReviewsController.java [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+//    Map<String, Review> persistedReview = new HashMap<>();
+//    review.setUserName(newReview.get("name"));
+//    review.setRating(Integer.valueOf(newReview.get("rating")));
+//    review.setReview(newReview.get("review"));
+//    review.setLongestRelationship(Integer.valueOf(newReview.get("longestRelationship")));
+//    reviewService.save(review);
+//    persistedReview.put("review", review);
+//
+//    return persistedReview;
+//  }
```
Comment (safang99) [inline src/main/java/com/launchacademy/reviews/controllers/ApiV1ReviewsController.java]: If the form is currently unfinished and this is a PR just for displaying reviews, then I would remove the commented out code and the review form from this PR to keep it scoped properly

File: src/main/java/com/launchacademy/reviews/controllers/ApiV1ReviewsController.java (hunk 2 above)
Comment (fbraese) [reply src/main/java/com/launchacademy/reviews/controllers/ApiV1ReviewsController.java]: Yes, this is for a form.

File: src/main/java/com/launchacademy/reviews/models/Review.java (hunk 3 above)
Comment (fbraese) [reply src/main/java/com/launchacademy/reviews/models/Review.java]: Yes, this is true & has been brought back.

**Review by safang99 — CHANGES_REQUESTED** [review]
Great job and it looks like there is one complete PR in here along with part of a second.  Currently I see a PR for creating reviews and having them display on a `site` show page.  There also seems to be part of adding a `reviews form` which should be it's own PR.  Other than that just minor formatting issues.

File: src/main/java/com/launchacademy/reviews/seeders/ReviewSeeder.java (hunk 4 above)
Comment (fbraese) [reply src/main/java/com/launchacademy/reviews/seeders/ReviewSeeder.java:262]: Yes, our next story is to add a form component on each site show page to add reviews.  This story is just to seed and display reviews.

File: src/main/frontend/components/NewReviewForm.js (hunk 5 above)
Comment (Christiandecolombia) [reply src/main/frontend/components/NewReviewForm.js]: Spacing is odd, We use Intellj to format the document.

Comment (Christiandecolombia) [reply src/main/frontend/components/NewReviewForm.js]: Updated setting to 2 spaces

File: src/main/frontend/components/SitesIndex.js (hunk 6 above)
Comment (Christiandecolombia) [reply src/main/frontend/components/SitesIndex.js]: removed

File: src/main/java/com/launchacademy/reviews/controllers/ApiV1ReviewsController.java (hunk 8 above)
Comment (Christiandecolombia) [reply src/main/java/com/launchacademy/reviews/controllers/ApiV1ReviewsController.java]: Commented out code has been removed

File: src/main/frontend/components/SitesIndex.js (hunk 7 above)
Comment (fbraese) [reply src/main/frontend/components/SitesIndex.js]: Great note!  We meant to update this and just have.  Thank you

**Pull request merged**